packages/suite/src/actions/suite/storageActions.ts

```diff
@@ -81,6 +82,57 @@ export const saveAccounts = async (accounts: Account[]) => {
     return db.addItems('accounts', accounts, true);
 };
 
+interface AccountPart {
+    symbol: Account['symbol'];
+    accountType: Account['accountType'];
+    accountIndex: Account['index'];
+    descriptor: Account['descriptor'];
+}
+
+export const saveBuyTrade = async (buyTrade: BuyTrade, account: AccountPart, date: string) => {
+    return db.addItem(
+        'coinmarketTrades',
+        {
+            key: buyTrade.paymentId,
+            tradeType: 'buy',
+            date,
+            data: buyTrade,
+            account: {
+                descriptor: account.descriptor,
+                symbol: account.symbol,
+                accountType: account.accountType,
+                accountIndex: account.accountIndex,
+            },
+        },
+        undefined,
+        true,
+    );
+};
+
+export const saveExchangeTrade = async (
+    exchangeTrade: ExchangeTrade,
+    account: AccountPart,
+    date: string,
+) => {
+    return db.addItem(
+        'coinmarketTrades',
+        {
+            key: exchangeTrade.orderId,
+            tradeType: 'exchange',
+            date,
+            data: exchangeTrade,
+            account: {
```

Comment:
unnecessary re-creation on an account object

Reply:
It is storing only limited information about the account to the DB.